__tests__/gui/common_cy_functions/general.js

```diff
@@ -134,29 +135,50 @@ function expandPanel(panel) {
     const panelToBeExpanded = allPanels.filter((p) => {
         return p.panel === panel
     })[0];
-    const nonSelectedPanels = allPanels.splice(allPanels.indexOf(panelToBeExpanded.panel), 1);
-
-    let closedNoPanels = true;
-    nonSelectedPanels.forEach(nsp => {
-            if (closedNoPanels) {
-                cy.get(nsp.panel).then($p => {
-                    if ($p.is(':visible')) {
-                        cy.get(`${nsp.wrapper} ${panelToBeExpanded.button}`).click({force: true});
-                        closedNoPanels = false;
-                    }
-                })
-            }
-        }
-    )
 
-    if (closedNoPanels) {
-        //nothing was expanded
-        cy.get(`.panel-toggles ${panelToBeExpanded.button}`).click({force: true});
-    }
+    const nonSelectedPanels = allPanels.filter((p) => {
+        return p.panel !== panel
+    });
+
+    recTogglePanel(panelToBeExpanded, nonSelectedPanels, 0).then(function (result) {
+        if (result) {
```

Comment:
Wouldn't `recTogglePanel` always return a truthy for `result`?

Reply:
nope - if all the panels are already closed `recTogglePanel` returns false
https://github.com/OpenUpSA/wazimap-ng-ui/blob/506b442e77c3e11a25d939165b14bbded4a05ce4/__tests__/gui/common_cy_functions/general.js#L168